horace_core/sqw/@sqw/mask_pixels.m

```diff
@@ -51,27 +53,25 @@
         error('HORACE:sqw:invalid_argument', ...
             'If the mask object is a Horace object if must be sqw-type i.e. contain pixel information')
     end
-
-    if ~isequal(obj.pix.num_pixels, keep_info.pix.num_pixels)
+    if obj.pix.num_pixels ~= keep_info.pix.num_pixels
         error('HORACE:sqw:invalid_argument', ...
             'Number of pixels in input and mask object must be the same')
     end
 elseif (isnumeric(keep_info) || islogical(keep_info)) && ...
         numel(keep_info)~=numel(obj.data.s)
-    if ~islogical(keep_info)
-        if isscalar(keep_info)
-            if keep_info<0
-                error('HORACE:sqw:invalid_argument', ...
-                    'Can not mask all pixels. Num pixels to mask should be positive number. Provided: %d', ...
-                    keep_info);
-            elseif keep_info>obj.num_pixels
-                error('HORACE:sqw:invalid_argument', ...
-                    'Can not mask more pixels then available. Num pixels to mask should be smaller or equal to num_pixels (%d). Provided: %d', ...
-                    obj.num_pixels,keep_info)
-            end
-        else
-            keep_info=logical(keep_info);
-        end
+    num_keep = sum(keep_info);
```

Comment:
why? This routine (this branch) accepts mask for every pixel. Other masks processed by other branches/algorithm wrappers.  
I believe this change does not modify functionality and you were too, as its your suggested change implemented.

Reply:
The new function docstring suggests that any value >=1 is a valid `keep_info` flag. Hence if I supply a `keep_info` of `[2 15 5 1]` commensurate with a 4 pixel `sqw` according to the docstring this should work. But this `num_keep` will be set to `23`

Reply:
And the checks will fail as you trying to retain 23 pixels out of 4, and this is correct  because if you are providing such mask, you probably meant something else.

There is of course point that unit test should cover the checks failures and when I write an algorithm, I an writing this checks. But my ticket is not about improving or harmonizing the unit tests for algorithms -- I am enabling correct and unified filebacked operations. 
I can not fix all bugs within this particular ticket, though I am trying to fix what I see.